### PR TITLE
removing unecessary lru cache for gossip messages in p2p

### PR DESCRIPTION
**Pull request by svv232**
Explain your changes:
* Removing redundant caching in the lib2p networking layer. 
* This change deletes the recently_seen field from the `Resource_pool.t` in transaction_pool.ml. Now the daemon will solely rely on libp2p's networking layer to prevent reprocessing gossip messages.

Explain how you tested your changes:
*  ran `./scripts/mina-local-network/mina-local-network.sh --archive --update-genesis-timestamp --value-transfer-txns --zkapp-transactions` to check build and verified block production.

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [x] Does this close issues? List them
-  #11704

### CODE REVIEW

**Comment by svv232** [comment]
!ci-build-me

**Comment by dkijania** [comment]
@svv232 please run nightly on this pr (using !ci-nightly-me command) just make sure everything is ok on integration tests

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-nightly-me

File: src/lib/network_pool/transaction_pool.ml [hunk 1]
```diff
@@ -1106,20 +1083,7 @@ struct
...
         let open Intf.Verification_error in
-        let%bind () =
-          (* TODO: we should probably remove this -- the libp2p gossip cache should cover this already (#11704) *)
-          let (`Already_mem already_mem) =
-            Lru_cache.add t.recently_seen (Lru_cache.T.hash diff.data)
-          in
-          Deferred.return
-          @@ Result.ok_if_true
-               ((not already_mem) || is_sender_local)
-               ~error:Recently_seen
```
Comment (mrmr1993) [inline src/lib/network_pool/transaction_pool.ml:1121]: If we don't also use it elsewhere, it would be good to remove this error too (unless this error propagates over the protocol boundary to other nodes). Can be another PR though.

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/network_pool/transaction_pool.ml (hunk 1 above)
Comment (svv232) [reply src/lib/network_pool/transaction_pool.ml:1121]: removed it!

**Comment by svv232** [comment]
!ci-nightly-me

**Comment by svv232** [comment]
!ci-build-me

**Comment by svv232** [comment]
!ci-build-me

**Comment by svv232** [comment]
!ci-build-me

**Comment by svv232** [comment]
!ci-build-me
